langchain/spark_sql.py

```diff
@@ -0,0 +1,111 @@
+from typing import Any, Iterable, List, Optional
+
+from pyspark.sql import DataFrame, Row, SparkSession
+
+from langchain import SQLDatabase
+
+
+class SparkSQL(SQLDatabase):
```

Comment:
im guessing we don't want this to inherit from SQLDatabase since it's overwriting a lot of functionality and ignoring some functionality. @hwchase17 would it make sense to create an abstract SQLMixin that defines the main interface (get_table_info, run, from_uri, other public methods) and that both SparkSQL and SQLDatabase implement?

Reply:
@dev2049 The thing is that `SQLDatabaseToolkit` requires a `SQLDatabase`.  Inheriting from SQLDatabase can make it possible to use `SQLDatabaseToolkit` for Spark SQL.

Reply:
Double checked SQLDatabase, TBH, it’s better to rename it to `SQLAlchemyDatabase` instead of a generic name `SQLDatabase`. SparkSQL rewrote all most all of the functions. Maybe better to migrate it to its own class and add its own toolkit IMHO?

Reply:
@skcoirz yes I am ok with that.  Spark can verify the SQL query by itself so that it can have its own `QueryCheckerTool`